### ISSUE TITLE
simd.h: fix compilation with GCC < 4.4 (and llvm-gcc)

### ISSUE THREAD

**Issue by devernay** [OpenImageIO/oiio#1363]
### Problem

The fix in https://github.com/lgritz/oiio/commit/0b6de163ccf488d047cd44651541aeec81ebdcd0 included immintrin.h instead of the full collection of includes for intrinsics headers.

However, immintrin.h was introduced with GCC 4.4, so this breaks compilation on OSX 10.6 with llvm-g++ (and probably CentOS5 too).

(sorry for bothering you with that legacy platform, but we're still trying to support it)

**Expected behavior:** [What you expected to happen]

For reference, here is the list of intrinsics in GCC 4.4.0:
```
gcc-4.4.0/gcc/config/arm/mmintrin.h
gcc-4.4.0/gcc/config/i386/ammintrin.h
gcc-4.4.0/gcc/config/i386/avxintrin.h
gcc-4.4.0/gcc/config/i386/bmmintrin.h
gcc-4.4.0/gcc/config/i386/emmintrin.h
gcc-4.4.0/gcc/config/i386/immintrin.h
gcc-4.4.0/gcc/config/i386/mmintrin.h
gcc-4.4.0/gcc/config/i386/nmmintrin.h
gcc-4.4.0/gcc/config/i386/pmmintrin.h
gcc-4.4.0/gcc/config/i386/smmintrin.h
gcc-4.4.0/gcc/config/i386/tmmintrin.h
gcc-4.4.0/gcc/config/i386/wmmintrin.h
gcc-4.4.0/gcc/config/i386/x86intrin.h
gcc-4.4.0/gcc/config/i386/xmmintrin.h
gcc-4.4.0/gcc/config/ia64/ia64intrin.h
```

and in 4.3.6 (notice the absence of `immintrin.h`, `avxintrin.h` and `w86intrin.h`):
```
gcc-4.3.6/gcc/config/arm/mmintrin.h
gcc-4.3.6/gcc/config/i386/ammintrin.h
gcc-4.3.6/gcc/config/i386/bmmintrin.h
gcc-4.3.6/gcc/config/i386/emmintrin.h
gcc-4.3.6/gcc/config/i386/mmintrin.h
gcc-4.3.6/gcc/config/i386/nmmintrin.h
gcc-4.3.6/gcc/config/i386/pmmintrin.h
gcc-4.3.6/gcc/config/i386/smmintrin.h
gcc-4.3.6/gcc/config/i386/tmmintrin.h
gcc-4.3.6/gcc/config/i386/xmmintrin.h
gcc-4.3.6/gcc/config/ia64/ia64intrin.h
```

I made a fix available at https://raw.githubusercontent.com/MrKepzie/Natron/workshop/tools/MacOSX/ports/graphics/openimageio/files/oiio-1.6.10-simd.patch

It should probably be cleaned up from the #ifdef SSE stuff (including a header makes no harm after all).

Maybe there could be a configure-time chek for immintrin.h instead?

### Versions

* OIIO branch/version: 1.6.10
* OS: OSX 10.6
* C++ compiler: llvm-g++ 4.2

**Comment by lgritz** [comment]
Sorry about that. I'm ok with this patch.

**Comment by lgritz** [comment]
Your patch said:

    #if !defined(__GNUC__) || (OIIO_GNUC_VERSION > 40400 || __clang__)

If it was introduced with gcc 4.4, this could be `>= 40400`, right?

**Comment by devernay** [comment]
right! (I checked in the gcc source distributions one by one to find out when it was introduced)

**Comment by lgritz** [comment]
Merged into RB-1.6.

I'm not going to bother porting to master, since by being designated as eventually needing C++11, we don't need to worry about gcc < 4.4. (In fact, we don't need to worry about gcc < 4.8 for master.)
